src/python/pants/engine/internals/specs_rules.py

```diff
@@ -449,62 +446,9 @@ def __init__(
         )
 
 
-def _handle_ambiguous_result(
-    request: TargetRootsToFieldSetsRequest,
-    result: TargetRootsToFieldSets,
-    field_set_to_default_to: type[FieldSet],
-) -> TargetRootsToFieldSets:
-    assert len(result.targets) > 1
-    field_set_types = [type(field_set) for field_set in result.field_sets]
-
-    # NB: See https://github.com/pantsbuild/pants/pull/15849. We don't want to break clients as
-    # we shift behavior, so we add this temporary hackery here.
-    if (
-        # (We check for 2 targets because 3 would've been ambiguous pre-our-change)
-        len(result.targets) == 2
-        and PexBinaryFieldSet in field_set_types
-        and PythonSourceFieldSet in field_set_types
-    ):
-        if field_set_to_default_to is PexBinaryFieldSet:
-            warn_or_error(
-                "2.14.0dev1",
```

Comment:
So this is confusing, my apologies.

I'm thinking this should've been 2.15...

- In 2.13 we introduced the ambiguity, along with forcing our users to set the flag to choose behavior
- In 2.14, now the flag defaults to using the new behavior. But if the user still has it set to "use the old behavior" we should warn but honor that. There's still ambiguity.
- In 2.15 there's no choice, and therefore no ambiguity

Reply:
I'll post pr

Reply:
Excellent, thank you. I'll close this out.